vertical-pod-autoscaler/pkg/recommender/model/cluster.go

```diff
@@ -292,13 +292,23 @@ func (cluster *ClusterState) findOrCreateAggregateContainerState(containerID Con
 }
 
 // GarbageCollectAggregateCollectionStates removes obsolete AggregateCollectionStates from the ClusterState.
+// AggregateCollectionState is obsolete in two situations:
+// 1) It has no samples and there are no more active pods that can contribute,
+// 2) The last sample is too old to give meaningful recommendation.
```

Comment:
// 3) There are no samples and the container aggregator was created >8 days ago.

?

Reply:
Done